powershell-adapter/powershell.resource.ps1

```diff
@@ -1,427 +1,349 @@
         }
 
-        $z = [pscustomobject]@{
-            type = $fullResourceTypeName;
-            kind = 'Resource';
-            version = $version_string;
-            capabilities = $capabilities;
-            path = $r.Path;
-            directory = $r.ParentPath;
-            implementedAs = $r.ImplementationDetail;
-            author = $author_string;
-            properties = $propertyList;
-            requires = $requiresString
+        $resourceCache += [resourceCache]@{
+            Type            = "$moduleName/$($dsc.Name)"
+            DscResourceInfo = $DscResourceInfo
         }
-
-        $z | ConvertTo-Json -Compress
     }
+    return $resourceCache
 }
-elseif ($Operation -eq 'Get')
-{
-    $result = @()
-
-    RefreshCache
+$resourceCache = Invoke-CacheRefresh
+
+# Convert the INPUT to a configFormat object so configuration and resource are standardized as moch as possible
+function Get-ConfigObject {
+    param(
+        [Parameter(Mandatory = $true, ValueFromPipeline = $true)]
+        $stdinput
+    )
+    # normalize the INPUT object to an array of configFormat objects
+    $inputObj = $stdInput | ConvertFrom-Json -Depth 10
+    $desiredState = @()
+
+    # catch potential for improperly formatted configuration input
+    if ($inputObj.resources -and -not $inputObj.metadata.'Microsoft.DSC'.context -eq 'configuration') {
+        Write-Warning 'The input has a top level property named "resources" but is not a configuration. If the input should be a configuration, include the property: "metadata": {"Microsoft.DSC": {"context": "Configuration"}}'
+    }
 
-    if (IsConfiguration $inputobj_pscustomobj) # we are processing a config batch
-    {
-        foreach($r in $inputobj_pscustomobj.resources)
-        {
-            #Write-Output $r.type
-            $cachedResourceInfo = $script:ResourceCache[$r.type]
-            if ($cachedResourceInfo)
-            {
-                $inputht = @{}
-                $typeparts = $r.type -split "/"
-                $ModuleName = $typeparts[0]
-                $ResourceTypeName = $typeparts[1]
-                $r.properties.psobject.properties | %{ $inputht[$_.Name] = $_.Value }
-                $e = $null
-                $op_result = Invoke-DscResource -Method Get -ModuleName $ModuleName -Name $ResourceTypeName -Property $inputht -ErrorVariable e
-                if ($e)
-                {
-                    # By this point Invoke-DscResource already wrote error message to stderr stream,
-                    # so we just need to signal error to the caller by non-zero exit code.
-                    exit 1
-                }
-                $result += $op_result
-            }
-            else
-            {
-                $errmsg = "Can not find type " + $r.type + "; please ensure that Get-DscResource returns this resource type"
-                Write-Error $errmsg
-                exit 1
+    if ($inputObj.metadata.'Microsoft.DSC'.context -eq 'configuration') {
```

Comment:
It's a bit tedious, but you need to check each nested object isn't $null before you use it otherwise the script will have an exception

```suggestion
    if ($null -ne $inputObject.metadata -and $null -ne $inputObj.metadata.'Microsoft.DSC' -and $inputObj.metadata.'Microsoft.DSC'.context -eq 'configuration') {
```

Reply:
good suggestion. fixed.